server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -944,9 +944,13 @@ func (e *EventChannel) handleEvent(event *pepb.PublishBuildToolEventStreamReques
 		return nil
 	}
 
-	seqNo := event.OrderedBuildEvent.SequenceNumber
-	streamID := event.OrderedBuildEvent.StreamId
-	iid := streamID.InvocationId
+	if event.GetOrderedBuildEvent() == nil {
```

Comment:
can this happen?

Reply:
In a typical Bazel-created event stream, no. But in a hand-crafted event stream, yes.

(This is where we got the panic alert internally).

> Should we just proceed with a nil value for `ordered_build_event`?

That would make seqNo == 0, streamID == nil, and iid == "", which is pretty pointless.

So I think it's best if we just end the invalid stream early here.